utils/merge_test.go

```diff
@@ -0,0 +1,1648 @@
+	}
+
+	mergeStructEmbed2D = testStructEmbed2{1, 2, 3, 4, 5, 1.1, 2.2, "test", 10, 11, 12, 12, 13,
+		newInt(14), newInt8(15), newInt16(16), newInt32(17), newInt64(18),
+		newFloat64(19.9), newFloat32(20.1), newString("test pointer"),
+		newUint(21), newUint8(22), newUint16(23), newUint32(24), newUint64(25),
+		[]string{"test", "slice", "strings"}, []int{1, 2, 3, 4}, []float64{1.1, 2.2, 3.3},
+		map[string]int{"this": 1, "is": 2, "a": 3, "map": 4}, map[int]string{1: "this", 2: "is", 3: "another"},
+		map[string]*int{"wow": newInt(1), "a map": newInt(2), "of pointers!": newInt(3)},
+		map[int]*string{1: newString("Another"), 2: newString("map of"), 3: newString("pointers, wow!")},
+	}
+
+	mergeStructEmbedBaseA = testStructEmbed{1, 2, 3, 4, 5, 1.1, 2.2, "test", 10, 11, 12, 12, 13,
+		newInt(14), newInt8(15), newInt16(16), newInt32(17), newInt64(18),
+		newFloat64(19.9), newFloat32(20.1), newString("test pointer"),
+		newUint(21), newUint8(22), newUint16(23), newUint32(24), newUint64(25),
+		[]string{"test", "slice", "strings"}, []int{1, 2, 3, 4}, []float64{1.1, 2.2, 3.3},
+		map[string]int{"this": 1, "is": 2, "a": 3, "map": 4}, map[int]string{1: "this", 2: "is", 3: "another"},
+		map[string]*int{"wow": newInt(1), "a map": newInt(2), "of pointers!": newInt(3)},
+		map[int]*string{1: newString("Another"), 2: newString("map of"), 3: newString("pointers, wow!")},
+		mergeStructEmbed2A, &mergeStructEmbed2B,
+	}
+
+	mergeStructEmbedBaseB = testStructEmbed{1, 2, 3, 4, 5, 1.1, 2.2, "test", 10, 11, 12, 12, 13,
+		newInt(14), newInt8(15), newInt16(16), newInt32(17), newInt64(18),
+		newFloat64(19.9), newFloat32(20.1), newString("test pointer"),
+		newUint(21), newUint8(22), newUint16(23), newUint32(24), newUint64(25),
+		[]string{"test", "slice", "strings"}, []int{1, 2, 3, 4}, []float64{1.1, 2.2, 3.3},
+		map[string]int{"this": 1, "is": 2, "a": 3, "map": 4}, map[int]string{1: "this", 2: "is", 3: "another"},
+		map[string]*int{"wow": newInt(1), "a map": newInt(2), "of pointers!": newInt(3)},
+		map[int]*string{1: newString("Another"), 2: newString("map of"), 3: newString("pointers, wow!")},
+		mergeStructEmbed2C, &mergeStructEmbed2D,
+	}
+
+	expectedMerged = testStruct{1, 2, 3, 4, 5, 1.1, 2.2, "test", 10, 11, 12, 12, 13,
+		newInt(14), newInt8(15), newInt16(16), newInt32(17), newInt64(18),
+		newFloat64(19.9), newFloat32(20.1), newString("test pointer"),
+		newUint(21), newUint8(22), newUint16(23), newUint32(24), newUint64(25),
+		[]string{"test", "slice", "strings"}, []int{1, 2, 3, 4}, []float64{1.1, 2.2, 3.3},
+		map[string]int{"this": 1, "is": 2, "a": 3, "map": 4}, map[int]string{1: "this", 2: "is", 3: "another"},
+		map[string]*int{"wow": newInt(1), "a map": newInt(2), "of pointers!": newInt(3)},
+		map[int]*string{1: newString("Another"), 2: newString("map of"), 3: newString("pointers, wow!")},
+		mergeStructEmbedBaseA, &mergeStructEmbedBaseB,
+	}
+
+}
+
+func mergeSimple(base, patch simple) (*simple, error) {
```

Comment:
Heh, this is proof of why Go needs /something/ for generics. :)

Reply:
No kidding. But, I'm starting to appreciate that here I get more control, and it's perfectly clear what's happening. :)